cli/flox-watchdog/src/process.rs

```diff
@@ -0,0 +1,139 @@
+use std::fs::read_to_string;
+
+use tracing::debug;
+
+/// https://man7.org/linux/man-pages/man5/proc_pid_stat.5.html
+/// Process has finished but the parent hasn't reaped it yet.
+const PROCFS_STATE_ZOMBIE: &str = "Z";
+/// Process is dead and will transition to a zombie or disappear.
+/// Technically we shouldn't see this, but just in case:
+/// https://unix.stackexchange.com/a/653370
+const PROCFS_STATE_DEAD: &str = "X";
+const PROCFS_STATE_DEAD_COMPAT: &str = "x";
+
+/// Process watcher for Linux based on `procfs`.
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) struct ProcfsWatcher {
+    pid: i32,
+}
+
+/// Allow compilation when not on Linux to make IDE/LSP development easier. It
+/// will always return negative results when not used on other platforms.
+#[cfg_attr(not(target_os = "linux"), allow(dead_code))]
+impl ProcfsWatcher {
```

Comment:
suggestion (nb): put into a submodule and cfg that

Reply:
Done. There was a discussion about if we could make this easier to develop on MacOS but I don't think it's worth spending more time on so I'll settle for temporarily commenting the `cfg` lines when needed: https://flox-dev.slack.com/archives/C05P6A5J6U8/p1727434011859509